kudu/kudu.py

```diff
@@ -0,0 +1,73 @@
+import urllib2
+import json
+import argparse
+
+
+def main(servers):
+    metrics = []
+    for i in range(len(servers)):
+        server = servers[i]
+        url = "{}/metrics".format(server)
+        response = urllib2.urlopen(url)
+        data = json.loads(response.read())
+        metrics_to_append = parse_metrics(data)
+        for m in metrics_to_append:
+            metrics.append(m)
+
+    print(json.dumps(metrics, ensure_ascii=False))
+
+
+def parse_metrics(all_metrics):
+    metrics_to_return = []
+
+    for item in all_metrics:
+        metrics_list = item.get("metrics")
+
+        if item.get("type") == "server":
+            metrics = {}
+            prefix = item.get("id")
+            prefix = prefix[5:]
```

Comment:
Add a comment above this line explaining what we are doing here, possibly with an example id.

Reply:
Also again, try to account for unexpected failure scenarios (missing dict keys etc) throughout.